latch_cli/snakemake/single_task_snakemake.py

```diff
@@ -291,5 +295,72 @@ def get_wildcards(self, requested_output, wildcards_dict=None):
 
 RRule.get_wildcards = get_wildcards
 
+
+def get_docker_cmd(
+    img_path: str,
+    cmd: str,
+    args: str = "",
+    quiet: bool = False,
+    envvars: Dict[str, str] = None,
+    shell_executable: str = None,
+    container_workdir: str = None,
+    is_python_script: bool = False,
+):
+    if shell_executable is None:
+        shell_executable = "sh"
+    else:
+        # Ensure to just use the name of the executable, not a path,
+        # because we cannot be sure where it is located in the container.
+        shell_executable = Path(shell_executable).name
+
+    workdir = container_workdir if container_workdir is not None else "/latch"
+
+    docker_cmd = ["docker", "run"]
+    if envvars is not None:
+        for k, v in envvars.items():
+            docker_cmd.extend(["--env", f"{k}={v}"])
+
+    # support for containers currently only works with relative input and output
+    # paths. This preserves the behavior before the monkey path
+    docker_cmd.extend(
+        [
+            "--workdir",
+            f"{workdir}",
+            "--mount",
+            f'type=bind,src=.,"target={workdir}"',
+        ]
+    )
+
+    if args != "":
+        docker_cmd.extend(shlex.split(args))
+
+    docker_cmd.extend(
+        [
+            img_path,
+            shell_executable,
+            "-c",
+            cmd.replace("'", '"'),
```

Comment:
what is this replace for? seems like you might want to use `shlex.quote` here instead

Reply:
you are right, shlex join will automatically quote the `cmd` so no need to do anything here.